packages/terra-data-grid/src/subcomponents/ColumnHeaderCell.jsx

```diff
@@ -153,10 +196,28 @@ const ColumnHeaderCell = (props) => {
     switch (key) {
       case KeyCode.KEY_SPACE:
       case KeyCode.KEY_RETURN:
-        onColumnSelect(id, { row: rowIndex, col: columnIndex });
+        if (isSelectable && onColumnSelect) {
+          onColumnSelect(id, { row: rowIndex, col: columnIndex }, isSelectable);
+        }
         event.stopPropagation();
         event.preventDefault(); // prevent the default scrolling
         break;
+      case KeyCode.KEY_LEFT:
+        if (isResizable && isResizeHandleActive) {
+          columnHeaderCellRef.current.focus();
+          setResizeHandleActive(false);
+
+          event.stopPropagation();
+          event.preventDefault();
+        }
+        break;
+      case KeyCode.KEY_RIGHT:
+        if (isResizable && !isResizeHandleActive) {
```

Comment:
The _isResizeHandleActive_ is not getting reset if you tab in and out of the grid from the resizeHandler. 
As a result when you tab back into the grid and navigate right, resizeHandler is skipped and focus goes to the next cell. 
Not sure of the required behavior but when we tab back into the grid either
-  focus goes back to the resizeHandler and the right arrow then goes to the next cell or
- focus goes to the cell and the right arrow goes to the resizeHandler

Reply:
Fixed in 74fb315bfbf49d46e4c4fb4dfe7a02fb165df62b. Tabbing back will now return focus to the cell and right arrow will go to the resizeHandler. Currently working on adding tests for it.